tests/main/views/test_drafts.py

```diff
@@ -56,6 +62,500 @@ def service_count(self):
+
+        res = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+                **self.basic_questions_json,
+                'targetFramework': 'g-cloud-7'
+            }),
+            content_type='application/json')
+        assert res.status_code == 201
+
+        post_copy_source_service = Service.query.first()
+        assert post_copy_source_service.copied_to_following_framework is True
+
+    def test_source_service_is_not_marked_as_copied_if_copied_to_same_framework(self):
+        pre_copy_source_service = Service.query.first()
+        assert pre_copy_source_service.copied_to_following_framework is False
+
+        res = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+            }),
+            content_type='application/json')
+        assert res.status_code == 201
+
+        post_copy_source_service = Service.query.first()
+        assert post_copy_source_service.copied_to_following_framework is False
+
+    @pytest.mark.parametrize('framework_status', Framework.STATUSES)
+    def test_services_can_not_be_copied_to_a_framework_that_is_not_open(self, framework_status):
+        if framework_status == 'open':
+            return
+
+        self.set_framework_status('g-cloud-7', framework_status)
+
+        res = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+                **self.basic_questions_json,
+                'targetFramework': 'g-cloud-7'
+            }),
+            content_type='application/json')
+
+        assert res.status_code == 400
+        assert 'Target framework is not open' in res.get_data(as_text=True)
+
+
+class TestCopyPublishedFromFramework(DraftsHelpersMixin):
+    def post_to_copy_published_from_framework(
+        self, source_framework_slug="g-cloud-6", override_questions_to_copy=None, target_framework_slug='g-cloud-7'
+    ):
+        """
+        :param source_framework_slug: The framework slug for the services being copied.
+        :param override_questions_to_copy: A list of questions to copy, rather than the default fixture. There is a
+                                           special case where the string 'No questions' is used. This causes the
+                                           'questionsToCopy' key to be skipped from the data for test purposes.
+        :param target_framework_slug: The frameworks slug for where the drafts are being created.
+        :return: The json response from the api
+        """
+        data = {
+            **self.updater_json,
+            "sourceFrameworkSlug": source_framework_slug,
+            "supplierId": 1,
+        }
+
+        if override_questions_to_copy == 'No questions':
```

Comment:
It's still a bit dirty, but probably I'd be more comfortable having the special case of an empty list being passed in rather than a magic string. So we can do `if override_questions_to_copy == [] / elif override_questions_to_copy / else`

Reply:
Works for me.